management/src/main/java/org/kaazing/gateway/management/monitoring/configuration/impl/MonitorFileWriterImpl.java

```diff
@@ -0,0 +1,321 @@
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.kaazing.gateway.management.monitoring.configuration.impl;
+
+import java.io.File;
+import java.nio.ByteBuffer;
+import java.nio.ByteOrder;
+import java.nio.MappedByteBuffer;
+
+import org.kaazing.gateway.management.monitoring.configuration.MonitorFileWriter;
+import org.kaazing.gateway.management.monitoring.service.MonitoredService;
+import org.kaazing.gateway.management.monitoring.writer.GatewayWriter;
+import org.kaazing.gateway.management.monitoring.writer.ServiceWriter;
+import org.kaazing.gateway.management.monitoring.writer.impl.MMFGatewayWriter;
+import org.kaazing.gateway.management.monitoring.writer.impl.MMFSeviceWriter;
+import org.kaazing.gateway.service.MonitoringEntityFactory;
+
+import uk.co.real_logic.agrona.BitUtil;
+import uk.co.real_logic.agrona.concurrent.UnsafeBuffer;
+
+/**
+ * Class responsible for storing/writing information in the appropriate the MMF format.
+ *
+ * File layout:
+ * +-----------------------------------------------------------------------+
+ * | File version | GW data offset | Service mappings offset | | GW ID | GW counters
+ * lbl buffer offset | GW counters lbl buffer length | GW counters values buffer
+ * offset | GW counters values buffer length | | Number of services | Service 1 name
+ * | Service 1 offset | ... | Service 1 lbl buffer offset | Service 1 lbl buffer length |
+ * | Service 1 values buffer offset | Service 1 values buffer length | | ... |
+ * | GW counters labels buffer | | GW counters values buffer | | Service 1 labels buffer |
+ * | Service 1 values buffer || ... |
+ * +-----------------------------------------------------------------------+
+ * Metadata length: NUMBER_OF_INTS_IN_HEADER * BitUtil.SIZE_OF_INT + SIZEOF_STRING +
+ * servicesCount * (SIZEOF_STRING + NUMBER_OF_INTS_PER_SERVICE * BitUtil.SIZE_OF_INT)
+ */
+public final class MonitorFileWriterImpl implements MonitorFileWriter {
+    private static final int OFFSETS_PER_SERVICE = 4;
+    private static final int NUMBER_OF_INTS_PER_SERVICE = 5;
+    private static final int NUMBER_OF_INTS_IN_HEADER = 8;
+    private static final int SIZEOF_STRING = 128;
+    private static final int MAX_SERVICE_COUNT = 100;
+
+    private static final int MONITOR_VERSION = 1;
+    private static final int MONITOR_VERSION_OFFSET = 0;
+    private static final int GW_DATA_REFERENCE_OFFSET = MONITOR_VERSION_OFFSET + BitUtil.SIZE_OF_INT;
+    private static final int SERVICE_DATA_REFERENCE_OFFSET = GW_DATA_REFERENCE_OFFSET + BitUtil.SIZE_OF_INT;
+
+    private static final int GW_ID_OFFSET = SERVICE_DATA_REFERENCE_OFFSET + BitUtil.SIZE_OF_INT;
+    private static final int GW_DATA_OFFSET = GW_ID_OFFSET;
+
+    private static final int GATEWAY_COUNTER_VALUES_BUFFER_LENGTH = 1024 * 128;
```

Comment:
Wow, that's pretty big isn't it? I can't imagine us having that many gateway or service counters. I would think we could reduce it to say 50. Where did 1024 come from? Whatever then we have please make sure this unit test to prove that we will throw suitable exception if the value is exceeded (i.e. some service tries to add more counters that we can support).